service/src/main/kotlin/fi/espoo/evaka/specialdiet/SpecialDietController.kt

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2017-2021 City of Espoo
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.specialdiet
+
+import fi.espoo.evaka.Audit
+import fi.espoo.evaka.shared.auth.AuthenticatedUser
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.domain.EvakaClock
+import fi.espoo.evaka.shared.security.AccessControl
+import fi.espoo.evaka.shared.security.Action
+import org.springframework.web.bind.annotation.*
+
+data class JamixSpecialDiet(val modelId: Int, val fields: JamixSpecialDietFields)
+
+data class JamixSpecialDietFields(val dietName: String, val dietAbbreviation: String)
+
+fun cleanupJamixDietList(specialDietList: List<JamixSpecialDiet>): List<SpecialDiet> {
+    return specialDietList
+        .map {
+            SpecialDiet(
+                it.modelId,
+                cleanupJamixDietNameString(it.fields.dietName),
+                cleanupJamixDietAbbreviationString(it.fields.dietAbbreviation)
+            )
+        }
+        .filterNot { it.name.contains("POISTA") }
+        .filterNot { it.name.isEmpty() && it.abbreviation.isEmpty() }
+}
+
+fun cleanupJamixDietNameString(s: String): String {
+    return s.replace("tsekattava", "", true)
+        .replace("tsek", "", true)
+        .replace(Regex("\\s+"), " ")
+        .trim()
+}
+
+fun cleanupJamixDietAbbreviationString(s: String): String {
+    return s.replace("ätarkasta", "", true).replace(Regex("\\s+"), " ").trim()
```

Comment:
just double checking that there's no typo in "ätarkasta"

Reply:
No typo. My guess is that by prepending the abbreviation with "ä" all of these get sorted to the end of list in some Jamix system and do not clutter the list of actually used values.